src/backend/distributed/commands/index_pg_source.c

```diff
@@ -0,0 +1,195 @@
+/*-------------------------------------------------------------------------
+ *
+ * index_pg_source.c
+ *    Helper functions copy & pasted from PostgreSQL source code.
+ *    All the functions in this file is copied from
+ *    postgres/src/backend/commands/indexcmds.c
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "commands/defrem.h"
+#include "distributed/commands.h"
+#include "distributed/listutils.h"
+#include "mb/pg_wchar.h"
+
+
+/* *INDENT-OFF* */
+
+/*
+ * This function is copy & paste from Postgres source code:
+ * 		postgres/src/backend/commands/indexcmds.c
+ *
+ * Select the name to be used for an index.
+ *
+ * The argument list is pretty ad-hoc :-(
+ */
+char *
+ChooseIndexName(const char *tabname, Oid namespaceId,
```

Comment:
we can even use this function to support unnamed indexes?

Reply:
I was thinking of the same https://github.com/citusdata/citus/pull/3994/files#diff-2fb367404fd67f00cf27350700dce00dR48